packages/autocomplete-core/src/resolve.ts

```diff
@@ -56,15 +60,36 @@ type RequestDescriptionPreResolvedCustom<TItem extends BaseItem> = {
 
 export function preResolve<TItem extends BaseItem>(
   itemsOrDescription: TItem[] | TItem[][] | RequesterDescription<TItem>,
-  sourceId: string
+  sourceId: string,
+  state: AutocompleteState<TItem>
 ):
   | RequestDescriptionPreResolved<TItem>
   | RequestDescriptionPreResolvedCustom<TItem> {
   if (isRequesterDescription<TItem>(itemsOrDescription)) {
+    const contextParameters =
+      itemsOrDescription.requesterId === 'algolia'
+        ? Object.assign(
+            {},
+            ...Object.keys(state.context).map((key) => {
+              return (state.context[key] as Record<string, unknown>)
+                ?.queryParameters;
```

Comment:
I'm not sure, I feel like it could be a valid use case to have plugins set parameters in the future. Maybe it's not needed to documented it yet, but I don't think people would set a second-level `queryParameters` key and expect it to do nothing. Not sure though, as context isn't processed outside of this code

Reply:
I suggested `algoliaSearchParameters` in another comment.

I guess @FabienMotte's suggestion of using the `__` prefix might apply if we want to be extra careful and avoid sending unwanted parameters if existing customers happen to already use this key in custom plugins.